hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieHiveCatalog.java

```diff
@@ -656,12 +657,10 @@ public void dropTable(ObjectPath tablePath, boolean ignoreIfNotExists)
       client.dropTable(
           tablePath.getDatabaseName(),
           tablePath.getObjectName(),
-          // Indicate whether associated data should be deleted.
-          // Set to 'true' for now because Flink tables shouldn't have data in Hive. Can
-          // be changed later if necessary
-          true,
+          // External table drops only the metadata, should not delete the underlying data.
+          !TableType.EXTERNAL_TABLE.name().equals(getHiveTable(tablePath).getTableType().toUpperCase(Locale.ROOT)),
```

Comment:
This line should be reverted.

This should be set true because there is determine condition in HiveMetaStore impl whether deleting data by "deleteData" AND "is not external table" .
So set "deleteData" to true is expected.
```JAVA
boolean shouldDeleteData = deleteData && !isExternal;
```

Reply:
@Leoyzen, the `HiveMetaStore` of Hive 2.3 and 3.x doesn't delete data when table type is `EXTERNAL_TABLE` for [L1715](https://github.com/apache/hive/blob/branch-2.3/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java#L1715)
[L2482](https://github.com/apache/hive/blob/branch-3.0/standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java#L2482). This doesn't need to change for `EXTERNAL_TABLE`, therefore I have reverted this change.